src/core/net/netif.cpp

```diff
@@ -167,13 +159,6 @@ otError Netif::UnsubscribeAllRoutersMulticast(void)
 {
     otError error = OT_ERROR_NONE;
 
-    if (mMulticastAddresses == &kLinkLocalAllRoutersMulticastAddress)
```

Comment:
Same situation here. If the head of linked-list is pointing to `AllRouters`, to unsubscribe from All Routers address, we need to update the head to point to `AllNodes` entry.

Reply:
see above.